src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -440,12 +440,15 @@ public static void cloneFromScm(
         PluginSourcesUnavailableException lastException = null;
         for (String connectionURL : connectionURLs) {
             if (connectionURL != null) {
+                if (StringUtils.startsWith(connectionURL, "scm:git:")) {
+                    connectionURL = StringUtils.substringAfter(connectionURL, "scm:git:");
+                }
                 // See: https://github.blog/2021-09-01-improving-git-protocol-security-github/
                 connectionURL = connectionURL.replace("git://", "https://");
             }
             try {
                 cloneImpl(connectionURL, scmTag, checkoutDirectory);
-                break;
+                return; // checkout was ok
```

Comment:
also fixes a bug where if the first clone failed, and subsequent ones passed an exception would still be thrown as `lastException` would not be `null` further down when the loop exited.